tests/MongoDB.Driver.Tests/Linq/Linq3ImplementationTests/Serializers/KnownSerializers/KnownSerializersTests.cs

```diff
@@ -0,0 +1,137 @@
+/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver.Linq;
+using MongoDB.Driver.Linq.Linq3Implementation;
+using MongoDB.Driver.Linq.Linq3Implementation.Translators.ExpressionToExecutableQueryTranslators;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Linq.Linq3ImplementationTests.Serializers.KnownSerializers
+{
+    public class KnownSerializersTests
+    {
+        #region static
+        private static readonly IMongoDatabase __database = DriverTestConfiguration.Client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
+
+        static KnownSerializersTests()
+        {
+            BsonClassMap.RegisterClassMap<C3>(cm =>
+            {
+                cm.AutoMap();
+                cm.MapMember(p => p.Es).SetSerializer(new EnumSerializer<E>(BsonType.String));
+            });
+        }
+        #endregion
+
+        #region class definitions for testing
+        public enum E { A, B };
+
+        private class C1
+        {
+            public E Ei { get; set; }
+        }
+
+        private class C2
+        {
+            public E Ei { get; set; }
+            [BsonRepresentation(BsonType.String)]
+            public E Es { get; set; }
+        }
+
+        private class C3
+        {
+            public E Es { get; set; }
+        }
+
+        private class Results
+        {
+            public bool Result { get; set; }
+        }
+        #endregion
+
+        [Theory]
+        [InlineData(E.A, "{ \"Result\" : { \"$eq\" : [ \"$Ei\", 0 ] }, \"_id\" : 0 }")]
+        [InlineData(E.B, "{ \"Result\" : { \"$eq\" : [ \"$Ei\", 1 ] }, \"_id\" : 0 }")]
+        public void Where_operator_equal_should_render_correctly(E value, string expectedProjection)
+        {
+            var collection = __database.GetCollection<C1>(DriverTestConfiguration.CollectionNamespace.CollectionName);
+            var subject = collection.AsQueryable3();
```

Comment:
Rather than using an implied state coming from fields (either static or instance), use a helper method instead.

Replace lines 75-76 with:

    var subject = GetSubject<C1>();

Reply:
Where `GetSubject` would be:

    private IQueryable<TDocument> GetSubject<TDocument>()
    {
        var client = DriverTestConfiguration.Client;
        var database = client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
        var collection = database.GetCollection<TDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
        return collection.AsQueryable3();
    }

Reply:
Note:  we plan to add a new `DriverTestConfiguration.Linq3Client` property. When that exists the `GetSubject` helper method would instead become:

    private IQueryable<TDocument> GetSubject<TDocument>()
    {
        var client = DriverTestConfiguration.Linq3Client;
        var database = client.GetDatabase(DriverTestConfiguration.DatabaseNamespace.DatabaseName);
        var collection = database.GetCollection<TDocument>(DriverTestConfiguration.CollectionNamespace.CollectionName);
        return collection.AsQueryable(); // instead of AsQueryable3
    }

Reply:
Refactored as suggested.